defi-pulse/src/index-allocations/index.ts

```diff
@@ -0,0 +1,29 @@
+import { ethers } from 'ethers'
+import { util } from '@chainlink/ea-bootstrap'
+
+const SetABI = [
+  'function getComponents() external view returns (address[] memory)',
+  'function getUnits() external view returns (uint256[] memory)',
+]
+
+type Allocations = {
+  components: string[]
+  units: number[]
+}
+
+export const getAllocations = async (setAddress: string): Promise<Allocations> => {
```

Comment:
Maybe we can use the typed [setProtocol.js](https://github.com/SetProtocol/setprotocol.js/blob/a6f8a181c7229b62f602f2e78bdb3ccdaf55b1e0/src/api/SetTokenAPI.ts) lib directly for this.